### PR TITLE
add config.InstrumentAttributesGetter to optionally append attrs to `latency` instrument

### LINKED ISSUE #333: Support for query specific attributes in `latency` histogram
### Problem Statement
Hi! thanks for the awesome pkg. It has been quite useful for instrumenting sql calls.
We want to start leveraging metrics more instead of traces for creating our internal dashboards, given that traces may need to be sampled at some point.

There's a `db.sql.latency` histogram metric which would work. But it's missing a few attributes: database name, query operation name, table name etc.
The [semantic conventions for database metrics](https://opentelemetry.io/docs/specs/semconv/database/database-metrics/#metric-dbclientoperationduration) have the attributes documented but status is still "Experimental"

For traces, we have been able to reformat the span name using `SpanNameFormatter` and set more attributes using `AttributesGetter`. I was wondering if you'd open to supporting a similar `AttributesGetter` for metrics till status for metrics semconv is still Experimental? Happy to open a PR if this seems okay!

### Proposed Solution

Similar to `AttributesGetter` for adding more attributes in span based on query context here: https://github.com/XSAM/otelsql/blob/de82710d7a7913cae38a261b656ba68b5b9e59b4/utils.go#L96-L98

same or another attributes getter for metrics to append attributes to metrics here https://github.com/XSAM/otelsql/blob/de82710d7a7913cae38a261b656ba68b5b9e59b4/utils.go#L67

#### Alternatives

otelsql could add all the attributes documented [here](https://opentelemetry.io/docs/specs/semconv/database/database-metrics/#metric-dbclientoperationduration) by parsing queries, but:
* the spec is still experimental - attribute name can change till the spec becomes "Stable"
* otesql would need a lot of database specific query parsing logic which may not be right

#### Prior Art

Although not the same area, otelgrpc appends parsed rpc/context specific attributes to both spans & metrics:
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/3fab5f5f20fb7f8ee7e33172fd129d336cd5640d/instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go#L452-L462

https://github.com/open-telemetry/opentelemetry-go-contrib/blob/3fab5f5f20fb7f8ee7e33172fd129d336cd5640d/instrumentation/google.golang.org/grpc/otelgrpc/interceptor.go#L336-L337

### Additional Context

NA

### PR DESCRIPTION
**Pull request by krantideep95**
This allows setting custom attributes from query context onto latency metric, like this:
```go
otelsql.Register(databaseDriver,
	otelsql.WithInstrumentAttributesGetter(func(ctx context.Context, method otelsql.Method, query string, args []driver.NamedValue) []attribute.KeyValue {
	attrs := []attribute.KeyValue{
		semconv.DBNameKey.String(dbName),
	}

	operation, table := parseQuery(query)
	return append(attrs, semconv.DBSQLTableKey.String(table), semconv.DBOperationKey.String(operation))
  }),
)
```

resolves #333

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ type SpanNameFormatter func(ctx context.Context, method Method, query string) st
 // AttributesGetter provides additional attributes on spans creation.
 type AttributesGetter func(ctx context.Context, method Method, query string, args []driver.NamedValue) []attribute.KeyValue
 
+// InstrumentAttributesGetter provides additional attributes while recording metrics.
```
Comment (XSAM) [inline config.go:43]:
```suggestion
// InstrumentAttributesGetter provides additional attributes while recording metrics to instruments.
```

nit, as `instrument` is the concrete type for capturing metrics in OpenTelemetry. https://opentelemetry.io/docs/concepts/signals/metrics/#metric-instruments

Having mentioned this, the implementation is straightforward under the hood.

File: config.go [hunk 2]
```diff
@@ -73,6 +76,10 @@ type config struct {
 	// AttributesGetter will be called to produce additional attributes while creating spans.
 	// Default returns nil
 	AttributesGetter AttributesGetter
+
+	// InstrumentAttributesGetter will be called to produce additional attributes while recording metrics.
```
Comment (XSAM) [inline config.go:80]:
```suggestion
	// InstrumentAttributesGetter will be called to produce additional attributes while recording metrics to instruments.
```

Same as mentioned above.

File: option.go [hunk 3]
```diff
@@ -101,3 +101,10 @@ func WithAttributesGetter(attributesGetter AttributesGetter) Option {
 		cfg.AttributesGetter = attributesGetter
 	})
 }
+
+// WithInstrumentAttributesGetter takes InstrumentAttributesGetter that will be called every time metric is recorded.
```
Comment (XSAM) [inline option.go:105]:
```suggestion
// WithInstrumentAttributesGetter takes InstrumentAttributesGetter that will be called every time metric is recorded to instruments.
```

Same as mentioned above.

**Review by XSAM — COMMENTED** [review]
@krantideep95, thanks for the PR.

It looks good to me overall. Please also update the `CHANGELOG.md` file to reflect this new feature.

**Comment by krantideep95** [comment]
Thanks for the review @XSAM I have updated `CHANGELOG.md` and accepted your suggestions.

**Review by XSAM — APPROVED** [review]
@krantideep95 Thank you for the contribution!

**Pull request merged**